code/utils_model.py

```diff
@@ -1,479 +1,509 @@
+    """
+    Main function for training ResNet.
+    """
+    # Loading in the data.
+    data_transforms = get_data_transforms()
 
     image_datasets = {
-        x: datasets.ImageFolder(os.path.join(config.train_folder, x),
-                                data_transforms[x])
-        for x in ['train', 'val']
+        x: datasets.ImageFolder(root=str(config.args.train_folder.joinpath(x)),
+                                transform=data_transforms[x])
+        for x in ["train", "val"]
     }
+
     dataloaders = {
-        x: torch.utils.data.DataLoader(image_datasets[x],
-                                       batch_size=batch_size,
-                                       shuffle=(x is 'train'),
-                                       num_workers=4)
-        for x in ['train', 'val']
+        x: torch.utils.data.DataLoader(dataset=image_datasets[x],
+                                       batch_size=config.args.batch_size,
+                                       shuffle=(x is "train"),
+                                       num_workers=config.args.num_workers)
+        for x in ["train", "val"]
     }
-    dataset_sizes = {x: len(image_datasets[x]) for x in ['train', 'val']}
-
-    print(len(config.classes), 'classes:', config.classes)
-    print('num train images', len(dataloaders['train']) * batch_size)
-    print('num val images', len(dataloaders['val']) * batch_size)
-    print("CUDA is_available:", torch.cuda.is_available())
-    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
-
-    # initialize model
-    if resume_checkpoint is True:
-        model = torch.load(resume_checkpoint_path)
-        print('model loaded from', resume_checkpoint_path)
+    dataset_sizes = {x: len(image_datasets[x]) for x in ["train", "val"]}
+
+    print(
+        f"{len(config.classes)} classes: {config.classes}\n"
+        f"num train images {len(dataloaders['train']) * config.args.batch_size}\n"
+        f"num val images {len(dataloaders['val']) * config.args.batch_size}\n"
+        f"CUDA is_available: {torch.cuda.is_available()}")
+
+    model = create_model()
+    model = model.to(device=config.device)
+    optimizer = optim.Adam(params=model.parameters(),
+                           lr=config.args.learning_rate,
+                           weight_decay=config.args.weight_decay)
+    scheduler = lr_scheduler.ExponentialLR(
+        optimizer=optimizer, gamma=config.args.learning_rate_decay)
+
+    # Initialize the model.
+    if config.args.resume_checkpoint:
+        ckpt = torch.load(f=config.resume_checkpoint_path)
+        model.load_state_dict(state_dict=ckpt["model_state_dict"])
+        optimizer.load_state_dict(state_dict=ckpt["optimizer_state_dict"])
+        scheduler.load_state_dict(state_dict=ckpt["scheduler_state_dict"])
+        epoch = ckpt["epoch"]
+        print(f"model loaded from {config.resume_checkpoint_path}")
     else:
-        model = create_model(num_layers, pretrain)
+        epoch = 0
```

Comment:
what happened to created a model with pretraining?

Reply:
@jasonwei20 The create_model() function now checks if the pretraining is needed instead of passing in the pretrain argument.